pyxrf/model/command_tools.py

```diff
@@ -208,40 +219,179 @@ def pyxrf_batch(start_id, end_id=None, wd=None, fit_channel_sum=True, param_file
+            print(f"   data_file = {data_files}")
+            return 1
+
+        # At this point ``data_files`` is a list of str.
+        for n, fln in enumerate(data_files):
+            data_files[n] = os.path.expanduser(fln)
+
+        # Working directory name is appended to each file name in the list,
+        #     which does not contain full path
+        for n, fln in enumerate(data_files):
+            if not os.path.isabs(fln):  # Check if ``fln`` is an absolute file path
+                data_files[n] = os.path.join(wd, fln)
+
+        # Create the list of files. Include only existing files in the list.
         flist = []
-        for data_id in range(start_id, end_id+1):
-            flist += [fname for fname in all_files if str(data_id) in fname]
-        print('Number of files to fit: {}'.format(len(flist)))
-        print('\n'.join(flist))
+        for fln in data_files:
+            if os.path.exists(fln) and os.path.isfile(fln):
+                flist += [fln]
+            else:
+                print(f"WARNING: file {fln} does not exist.")
+
+    else:
+
+        # ``data_files`` parameter is None
+
+        all_files = glob.glob(os.path.join(wd, '*.h5'))
+
+        if start_id is None and end_id is None:
+            # ``start_id`` and ``end_id`` are not specified:
+            #   process all .h5 files in the current directory
+            flist = all_files
+        elif end_id is None:
+            # only ``start_id`` is specified:
+            #   process only one file that contains ``start_id`` in its name
+            #   (only if such file exists)
+            flist = [fname for fname in all_files
+                     if re.search(f"^[^_]*_{str(start_id)}\D+", fname)]  # noqa: W605
+            if len(flist) < 1:
+                print(f"File with Scan ID {start_id} was not found.")
+            else:
+                print(f"Processing file with Scan ID {start_id}")
+        else:
+            # ``start_id`` and ``end_id`` are specified:
+            #   select files, which contain the respective ID substring in their names
+            flist = []
+            for data_id in range(start_id, end_id+1):
+                flist += [fname for fname in all_files
+                          if re.search(f"^[^_]*_{str(data_id)}\D+", fname)]  # noqa: W605
+            if len(flist) < 1:
+                print(f"No files with Scan IDs in the range {start_id} .. {end_id} were not found.")
+            else:
+                print(f"Processing file with Scan IDs in the range {start_id} .. {end_id}")
+
+    # Check if .json parameter file exists
+    if not os.path.isabs(param_file_name):
+        pname = os.path.join(wd, param_file_name)
+    else:
+        pname = param_file_name
+    if not os.path.exists(pname) and not os.path.isfile(pname):
+        print(f"ERROR: can't find the parameter file {pname}. "
+              "Check the value of 'param_file_name'.")
+        return 1
```

Comment:
Again, I'd recommend to use `raise` here.

Reply:
Fixed